Release 2.13.2 (#10792)
PrefectHQ/prefect: RELEASE-NOTES.md

Use `enforce_parameter_schema` when deploying your flow to guard against invalid parameters:

```python
from prefect import flow
from pydantic import BaseModel

class Person(BaseModel):
    name: str
    greeting: str = "Hello"

@flow(log_prints=True)
def my_flow(person: Person, name: str = "world"):
    print(f'{person.name} says, "{person.greeting}, {name}!"')

if __name__ == "__main__":
    my_flow.serve(
        "testing-params",
        enforce_parameter_schema=True,
    )

```

An attempt to run the created deployment with invalid parameters will fail and give a reason the flow run cannot be created:
```bash
> prefect deployment run 'my-flow/testing-params' -p person='{"name": 1}'

Error creating flow run: Validation failed for field 'person.name'. Failure reason: 1 is not of type 'string'
```

You can enable parameter enforcement via `prefect deploy` with the `--enforce-parameter-schema` flag or by setting `enforce_parameter_schema` to `True` in your `prefect.yaml` file.

See the following pull request for details:
- https://github.com/PrefectHQ/prefect/pull/10773

> ### Enhanced Deployment Flexibility with Pattern-based Naming

Reviewer: ```suggestion
### Enhanced deployment flexibility with pattern-based naming
```

> ### Enhanced deployment flexibility with pattern-based naming

Reviewer: probably "deploying" not "naming" :D

> ### Enhanced deployment flexibility with pattern-based naming

Reviewer: ```suggestion
### Enhanced deployment flexibility with pattern-based deploying
```

### Documentation
- Update UI screenshot for role creation — https://github.com/PrefectHQ/prefect/pull/10732
- Add `push work pools` tag to push work pools guide to raise visibility — https://github.com/PrefectHQ/prefect/pull/10739
- Update docs with recent brand changes — https://github.com/PrefectHQ/prefect/pull/10736
- Update Prefect Cloud quickstart guide to include new features — https://github.com/PrefectHQ/prefect/pull/10742
- Fix broken diagram in workers tutorial — https://github.com/PrefectHQ/prefect/pull/10762
- Add screenshots to artifacts concept page — https://github.com/PrefectHQ/prefect/pull/10748
- Remove boost from block-based deployments page in documentation and improve visibility of `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10775
- Add example of retrieving default base job template to work pools concept documentation — https://github.com/PrefectHQ/prefect/pull/10784
- Add references to `enforce_parameter_schema` to docs — https://github.com/PrefectHQ/prefect/pull/10782
- Add documentation for pattern matching in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10791

> ## New contributors

Reviewer: ```suggestion
### New contributors
```

# Prefect Release Notes

## Release 2.13.2

### Server-side enforcement of deployment parameter schemas

> We've added the ability to enforce parameter schemas for deployments via the Prefect API! This feature will prevent creation of flow runs with parameters that are incompatible with deployed flow allowing you to discover errors sooner and avoid provisioning infrastructure for flow runs destined to fail.

Reviewer: ```suggestion
We've added the ability to enforce parameter schemas for deployments via the Prefect API! This feature will prevent creation of flow runs with parameters that are incompatible with deployed flows, allowing you to discover errors sooner and avoid provisioning infrastructure for flow runs destined to fail.
```

Use `enforce_parameter_schema` when deploying your flow to guard against invalid parameters:

```python
from prefect import flow
from pydantic import BaseModel

class Person(BaseModel):
    name: str
    greeting: str = "Hello"

@flow(log_prints=True)
def my_flow(person: Person, name: str = "world"):
    print(f'{person.name} says, "{person.greeting}, {name}!"')

if __name__ == "__main__":
    my_flow.serve(
        "testing-params",
        enforce_parameter_schema=True,
    )

```

An attempt to run the created deployment with invalid parameters will fail and give a reason the flow run cannot be created:
```bash
> prefect deployment run 'my-flow/testing-params' -p person='{"name": 1}'

Error creating flow run: Validation failed for field 'person.name'. Failure reason: 1 is not of type 'string'

> ```

Reviewer: Would it be weird to include an example of it in `prefect deploy`? e.g.

`prefect deploy --enforce-parameter-schema`

Author: I added a line about how to use parameter enforcement with `prefect deploy`, but let me know if you think an example is still needed.

Reviewer: Nice! Thanks

- Fixed command handling in `run_shell_script` deployment step on Windows — https://github.com/PrefectHQ/prefect/pull/10719
- Made `slot_decay_per_second` not nullable to fix validation issues — https://github.com/PrefectHQ/prefect/pull/10790
- Fixed Prefect variable resolution in deployments section of `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10783

### Documentation
- Updated UI screenshot for role creation — https://github.com/PrefectHQ/prefect/pull/10732
- Add `push work pools` tag to push work pools guide to raise visibility — https://github.com/PrefectHQ/prefect/pull/10739
- Update docs with recent brand changes — https://github.com/PrefectHQ/prefect/pull/10736
- Updated Prefect Cloud quickstart guide to include new features — https://github.com/PrefectHQ/prefect/pull/10742
- Fixed broken diagram in workers tutorial — https://github.com/PrefectHQ/prefect/pull/10762
- Added screenshots to Artifacts concept page — https://github.com/PrefectHQ/prefect/pull/10748
- Removed boost from block-based deployments page in documentation and improved visibility of `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10775

> 

Reviewer: ```suggestion
- Add documentation for work-pool get-default-base-job-template — https://github.com/PrefectHQ/prefect/pull/10784
- Adds references to `enforce_parameter_schema` to docs — https://github.com/PrefectHQ/prefect/pull/10782
- Add documentation for pattern matching in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10791
```

Use `enforce_parameter_schema` when deploying your flow to guard against invalid parameters:

```python
from prefect import flow
from pydantic import BaseModel

class Person(BaseModel):
    name: str
    greeting: str = "Hello"

@flow(log_prints=True)
def my_flow(person: Person, name: str = "world"):
    print(f'{person.name} says, "{person.greeting}, {name}!"')

if __name__ == "__main__":
    my_flow.serve(
        "testing-params",
        enforce_parameter_schema=True,
    )

```

An attempt to run the created deployment with invalid parameters will fail and give a reason the flow run cannot be created:
```bash
> prefect deployment run 'my-flow/testing-params' -p person='{"name": 1}'

Error creating flow run: Validation failed for field 'person.name'. Failure reason: 1 is not of type 'string'
```

You can enable parameter enforcement via `prefect deploy` with the `--enforce-parameter-schema` flag or by setting `enforce_parameter_schema` to `True` in your `prefect.yaml` file.

See the following pull request for details:
- https://github.com/PrefectHQ/prefect/pull/10773

### Enhancements

> - Added API route for work pool counts — https://github.com/PrefectHQ/prefect/pull/10770

Reviewer: fyi, we usually use the infinitive verb form for release notes, e.g. "Add"

- Added pattern matching for deployment name in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10772

> - Added command to get default base job template — https://github.com/PrefectHQ/prefect/pull/10776

Reviewer: to the CLI

### Fixes

> - On UI build, make paths relative rather than absolute.  — https://github.com/PrefectHQ/prefect/pull/10390

Reviewer: in `prefect dev build-ui` cmd

> - Lowered the upper bound on pinned pendulum library — https://github.com/PrefectHQ/prefect/pull/10752

Reviewer: ```suggestion
- Lower the upper bound on pinned `pendulum` library in `requirements.txt` — https://github.com/PrefectHQ/prefect/pull/10752
```

- Fixed command handling in `run_shell_script` deployment step on Windows — https://github.com/PrefectHQ/prefect/pull/10719

> - Made `slot_decay_per_second` not nullable to fix validation issues — https://github.com/PrefectHQ/prefect/pull/10790

Reviewer: Fix validation on concurrency limits?

# Prefect Release Notes

## Release 2.13.2

> ### Server-side enforcement of deployment parameter schemas

Reviewer: ```suggestion
### Opt-in to server-side enforcement of deployment parameter schemas
```
or alternatively, "This opt-in feature"

but I think it could be good to mention that

Use `enforce_parameter_schema` when deploying your flow to guard against invalid parameters:

```python
from prefect import flow
from pydantic import BaseModel

class Person(BaseModel):
    name: str
    greeting: str = "Hello"

@flow(log_prints=True)
def my_flow(person: Person, name: str = "world"):
    print(f'{person.name} says, "{person.greeting}, {name}!"')

if __name__ == "__main__":
    my_flow.serve(
        "testing-params",
        enforce_parameter_schema=True,
    )

```

An attempt to run the created deployment with invalid parameters will fail and give a reason the flow run cannot be created:
```bash
> prefect deployment run 'my-flow/testing-params' -p person='{"name": 1}'

Error creating flow run: Validation failed for field 'person.name'. Failure reason: 1 is not of type 'string'
```

You can enable parameter enforcement via `prefect deploy` with the `--enforce-parameter-schema` flag or by setting `enforce_parameter_schema` to `True` in your `prefect.yaml` file.

See the following pull request for details:
- https://github.com/PrefectHQ/prefect/pull/10773

### Enhancements
- Add API route for work pool counts — https://github.com/PrefectHQ/prefect/pull/10770

> - Add pattern matching for deployment name in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10772

Author: Yeah, a highlight would be good since I think there are a lot of users that have been asking for it. I'll add one based on the docs Serina wrote.

> - Add pattern matching for deployment name in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10772

Reviewer: writing one up quick!

> - Add pattern matching for deployment name in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10772

Reviewer: added one! not so quick haha

### Fixes
- Make paths relative rather than absolute in the `prefect dev build-ui` command — https://github.com/PrefectHQ/prefect/pull/10390
- Lower the upper bound on pinned pendulum library — https://github.com/PrefectHQ/prefect/pull/10752
- Fix command handling in `run_shell_script` deployment step on Windows — https://github.com/PrefectHQ/prefect/pull/10719
- Fix validation on concurrency limits — https://github.com/PrefectHQ/prefect/pull/10790
- Fix Prefect variable resolution in deployments section of `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10783

### Documentation
- Update UI screenshot for role creation — https://github.com/PrefectHQ/prefect/pull/10732
- Add `push work pools` tag to push work pools guide to raise visibility — https://github.com/PrefectHQ/prefect/pull/10739
- Update docs with recent brand changes — https://github.com/PrefectHQ/prefect/pull/10736
- Update Prefect Cloud quickstart guide to include new features — https://github.com/PrefectHQ/prefect/pull/10742
- Fix broken diagram in workers tutorial — https://github.com/PrefectHQ/prefect/pull/10762

> - Add screenshots to Artifacts concept page — https://github.com/PrefectHQ/prefect/pull/10748

Reviewer: ```suggestion
- Add screenshots to artifacts concept page — https://github.com/PrefectHQ/prefect/pull/10748
```

> - Remove boost from block-based deployments page in documentation and improved visibility of `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10775

Reviewer: ```suggestion
- Remove boost from block-based deployments page in documentation and improve visibility of `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10775
```

> - Add documentation for work-pool get-default-base-job-template — https://github.com/PrefectHQ/prefect/pull/10784

Reviewer: ```suggestion
- Add example of retrieving default base job template to work pools concept documentation — https://github.com/PrefectHQ/prefect/pull/10784
```
Maybe ?

- Add references to `enforce_parameter_schema` to docs — https://github.com/PrefectHQ/prefect/pull/10782
- Add documentation for pattern matching in `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10791

> ## New Contributors

Reviewer 2: ```suggestion
## New contributors
```

Reviewer 2: I see we had title case in the example below, so maybe not worth changing going forward, but the docs style guide has second level headings in second case, fwiw.
